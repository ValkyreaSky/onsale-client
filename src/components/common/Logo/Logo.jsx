import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
// Images
import logo from 'assets/navigation-icons/icon-logo.svg';

const LogoContainer = styled.div`
  display: flex;
  align-items: center;
`;

const LogoIcon = styled.svg`
  width: 3.8rem;
  margin-right: 1.3rem;
`;

const LogoBrand = styled(Link)`
  font-size: 2.2rem;
  color: inherit;
`;

class Logo extends Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <LogoContainer>
        <LogoIcon viewBox={logo.viewBox}>
          <use xlinkHref={logo} />
        </LogoIcon>
        <LogoBrand to="/">Selli</LogoBrand>
      </LogoContainer>
    );
  }
}

export default Logo;
export { LogoBrand };
