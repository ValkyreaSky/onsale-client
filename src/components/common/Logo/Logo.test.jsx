import React from 'react';
import { shallow } from 'enzyme';
import Logo, { LogoBrand } from './Logo';

describe('<Logo />', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render <LogoBrand> comopnent with a link to the homepage', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper.find(LogoBrand).length).toBe(1);
    expect(wrapper.find(LogoBrand).props().to).toBe('/');
  });

  it('should render <LogoBrand> comopnent with a link to the homepage', () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper.find(LogoBrand).props().to).toBe('/');
  });
});
