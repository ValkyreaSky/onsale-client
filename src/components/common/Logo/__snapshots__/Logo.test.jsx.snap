// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should match snapshot 1`] = `
<styled.div>
  <styled.svg>
    <use
      xlinkHref="icon-logo.svg"
    />
  </styled.svg>
  <Styled(Link)
    to="/"
  >
    Selli
  </Styled(Link)>
</styled.div>
`;
